File: apple/Sources/ReactNativeWatchOS/Modifiers/Opacity.swift
import SwiftUI

/// SwiftUI `.opacity(_:)`. Sets the transparency of the view (`0`…`1`).
enum RNWOpacityModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("opacity") { view, params, _ in
            AnyView(view.opacity(params.double("value") ?? 1))
        }
    }
}

File: src/modifiers/opacity.ts
import { createModifier } from './createModifier';

export interface OpacityParams {
  /// Opacity from `0` (fully transparent) to `1` (fully opaque).
  value: number;
}

/// SwiftUI `.opacity(_:)`. Sets the transparency of the view.
export function opacity(
  params: OpacityParams | number
): ReturnType<typeof createModifier> {
  if (typeof params === 'number') return createModifier('opacity', { value: params });
  return createModifier('opacity', params);
}
